addon/capabilities.js

```diff
@@ -0,0 +1,41 @@
+const CAPABILITIES = {
+  requestAnimationFrame: requestAnimationFrame,
+  requestIdleCallback: requestIdleCallback,
+};
+
+/**
+ * @type {Capabilities}
+ */
+let _instance = null;
+
+class Capabilities {
+  constructor() {
+    if (_instance) {
+      throw new Error('Capabilities can only be constructed once');
+    }
+
+    this.reset();
+    this.DISABLED = true;
```

Comment:
This is confusing, took me a few read throughs of the diff to grok what was even happening.

Do we really need it? Can't we call `.overrideCapabilities('requestIdleCallback', true)`?

Reply:
Of course we can. I added it for readability, but I think it's bad to have it as part of the class. I could export another var. Personally, I dislike booleans passed into methods as it doesn't do a good job informing people as to what the parameter controls.